OneSignalSDK/onesignal/src/main/java/com/onesignal/OSEmailSubscriptionState.java

```diff
@@ -62,7 +62,7 @@ void clearEmailAndId() {
     }
 
     void setEmailUserId(@NonNull String id) {
-        boolean changed = !id.equals(emailUserId);
+        boolean changed = id != null && !id.equals(emailUserId);
```

Comment:
`changed` should be `true` if setting email to `null` when it was set. This does not cover this. Let's use the same logic as `OSSubscriptionState.setUserId`.

Reply:
I will need to talk to you about this issue in general and what we should do to prevent a crash here instead.